### PR TITLE
Fix/merge release3.5 to3.6

### PR DESCRIPTION
**Pull request by sagarkapare**
Conflicts in pom.xml for cdap versions and in `UpgradeTool.java`

### CODE REVIEW

**Comment by anwar6953** [comment]
Why did UpgradeTool have conflict?

How did you resolve all the pom files?

**Comment by sagarkapare** [comment]
CDAP versions in the pom file was taken from release/3.6. UpgradeTool had conflict because rename of the DatasetManager to UpgradeDatasetManager

**Comment by sagarkapare** [comment]
Release build is running here - https://builds.cask.co/browse/CDAP-RUT179-1

**Comment by anwar6953** [comment]
Weren't there 100+ conflicts though? Just a manual replace?

**Comment by anwar6953** [comment]
What about in json files that have version in them?

**Comment by sagarkapare** [comment]
Yes. There were conflicts in multiple pom.xml and json about the CDAP version. I fixed them manually to have versions from release/3.6

File: cdap-distributions/src/hdinsight/install.sh [hunk 1]
```diff
@@ -22,9 +22,9 @@
 # The git branch to clone
 CDAP_BRANCH='release/3.5'
 # Optional tag to checkout - All released versions of this script should set this
-CDAP_TAG='v3.5.0'
+CDAP_TAG=''
 # The CDAP package version passed to Chef
-CDAP_VERSION='3.5.0-1'
+CDAP_VERSION='3.5.1-1'
```
Comment (anwar6953) [inline cdap-distributions/src/hdinsight/install.sh]: this needs to be updated?
@dereklwood

File: cdap-distributions/src/hdinsight/mainTemplate.json [hunk 2]
```diff
@@ -50,18 +50,18 @@
       },
       "installScriptActions": [{
         "name": "[concat('cdap-pageblob-configure-v0','-' ,uniquestring(variables('applicationName')))]",
-        "uri": "https://raw.githubusercontent.com/caskdata/cdap/release/3.5/cdap-distributions/src/hdinsight/pageblob-configure.sh",
+        "uri": "http://downloads.cask.co/hdinsight/pageblob-configure-3.5.1.sh",
```
Comment (anwar6953) [inline cdap-distributions/src/hdinsight/mainTemplate.json:53]: what about this

File: cdap-distributions/src/hdinsight/install.sh (hunk 1 above)
Comment (wolf31o2) [reply cdap-distributions/src/hdinsight/install.sh]: Yes.

File: cdap-distributions/src/hdinsight/mainTemplate.json (hunk 2 above)
Comment (wolf31o2) [reply cdap-distributions/src/hdinsight/mainTemplate.json:53]: Yes.

Comment (sagarkapare) [reply cdap-distributions/src/hdinsight/mainTemplate.json:53]: should this be updated to `http://downloads.cask.co/hdinsight/pageblob-configure-3.6.0.sh`? When is this script created?

Comment (sagarkapare) [reply cdap-distributions/src/hdinsight/mainTemplate.json:53]: updated to `"uri": "http://downloads.cask.co/hdinsight/pageblob-configure-3.6.0.sh"`

File: cdap-distributions/src/hdinsight/install.sh (hunk 1 above)
Comment (sagarkapare) [reply cdap-distributions/src/hdinsight/install.sh]: updated the version

**Comment by sagarkapare** [comment]
@anwar6953 @wolf31o2 Please help review the PR. Thanks! Build is green https://builds.cask.co/browse/CDAP-RUT179-1

**Comment by bdmogal** [comment]
Last commit addresses review comments correctly. LGTM

**Comment by sagarkapare** [comment]
Thanks @bdmogal. Merging.

File: cdap-distributions/src/hdinsight/install.sh (hunk 1 above)
Comment (wolf31o2) [reply cdap-distributions/src/hdinsight/install.sh]: Sorry. It needed to be updated to `3.5.1`, not to `3.6.0`. I thought that Ali was asking if that change was correct/necessary. It was correct as it was. It _must_ match a released version of CDAP.

File: cdap-distributions/src/hdinsight/mainTemplate.json (hunk 2 above)
Comment (wolf31o2) [reply cdap-distributions/src/hdinsight/mainTemplate.json:53]: This is incorrect. The script is created and published as part of the release. These files don't exist and won't exist until the release is published. That being said, we _will_ need to change them before the release is made, so this is the end-goal. It's just that currently, these scripts won't work, as is.

**Pull request merged**